apps/manager_app/auth_card.c

```diff
@@ -0,0 +1,542 @@
+                                            manager_auth_card_response_t *resp);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+auth_card_screen_ctx_t auth_card_ctx = {0};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static manager_error_code_t send_auth_card_response(
+    manager_auth_card_response_t *resp) {
+  if (NULL == resp || 0 == resp->which_response) {
+    return MANAGER_TASK_INVALID_ARGS;
+  }
+
+  // Initialize respose with zero values
+  manager_result_t result = MANAGER_RESULT_INIT_ZERO;
+
+  /* TODO: Update the buffer size when all option files have been updated for
+   * manager app*/
+  uint8_t buffer[MANAGER_AUTH_CARD_RESPONSE_SIZE + 4] = {0};
+  size_t bytes_written = 0;
+
+  // Set auth card valid tags
+  result.which_response = MANAGER_QUERY_AUTH_CARD_TAG;
+
+  // Copy response to pb result struct
+  memcpy(&(result.auth_card), resp, sizeof(manager_auth_card_response_t));
+
+  // Encode result to byte-stream
+  if (false ==
+      encode_manager_result(&result, buffer, sizeof(buffer), &bytes_written)) {
+    // Encoding failed, return false
+    return MANAGER_TASK_ENCODING_FAILED;
+  }
+
+  usb_send_msg(buffer, bytes_written);
+  return MANAGER_TASK_SUCCESS;
+}
+
+static void prepare_card_auth_context(
+    manager_auth_card_initiate_request_t *initiate_request) {
+  ASSERT(NULL != initiate_request);
+
+  auth_card_ctx.acceptable_cards = ACCEPTABLE_CARDS_ALL;
+
+  if (initiate_request->has_card_index) {
+    auth_card_ctx.acceptable_cards =
+        encode_card_number(initiate_request->card_index);
+    snprintf(auth_card_ctx.heading,
+             sizeof(auth_card_ctx.heading),
+             UI_TEXT_TAP_CARD,
+             tap_card_data.acceptable_cards);
+  } else {
+    snprintf(auth_card_ctx.heading,
+             sizeof(auth_card_ctx.heading),
+             ui_text_tap_a_card);
+  }
+
+  auth_card_ctx.pair_card_required = false;
+  if (initiate_request->has_pair_card) {
+    auth_card_ctx.pair_card_required = initiate_request->pair_card;
+  }
+  snprintf(auth_card_ctx.message,
+           sizeof(auth_card_ctx.message),
+           UI_TEXT_PLACE_CARD_TILL_BEEP,
+           SIGN_SERIAL_BEEP_COUNT(auth_card_ctx.pair_card_required));
+}
+
+bool wait_for_user_confirmation() {
+  confirm_scr_init(ui_text_start_verification_of_card);
+
+  evt_status_t status = get_events(EVENT_CONFIG_UI, MAX_INACTIVITY_TIMEOUT);
+
+  if (status.ui_event.event_occured) {
+    return status.ui_event.event_type == UI_EVENT_CONFIRM;
+  } else {
+    HANDLE_P0_EVENTS();
+  }
+  return false;
+}
```

Comment:
Let's remove this function for now, as it is not being used currently

Reply:
The purpose is to understand how we expect to handle the user confirmation, function call has been commented.
Conversely, we could make this function a generic one in a separate PR, and use it everywhere a confirmation is required.
Although that would be assuming that P0 events are handled in a generic way too.

Reply:
Yes. Let's take up confirmation and P0 separately.. and reduce work in this PR

Reply:
So please remove it for now

Reply:
removed